hpat/tests/tests_perf/test_perf_series_str.py

```diff
@@ -0,0 +1,383 @@
+
+def calc_results(f, test_data):
+    """Calculate time of boxing+unboxing and internal execution"""
+    test_iteration_number = 5
+
+    exec_times = []
+    boxing_times = []
+    for _ in range(test_iteration_number):
+        ext_start = time.time()
+
+        int_result = f(test_data)
+
+        ext_finish = time.time()
+        exec_times.append(int_result)
+        boxing_times.append(max(ext_finish - ext_start - int_result, 0))
+
+    return exec_times, boxing_times
+
+
+class TestSeriesStringMethods(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.test_results = TestResults()
+        cls.test_results.load()
+
+        cls.total_data_length = [10**4 + 513, 10**5 + 2025]
+        cls.width = [16, 64, 512, 1024]
+        cls.num_thread = int(os.environ.get('NUMBA_NUM_THREADS', config.NUMBA_NUM_THREADS))
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.test_results.print()
+        cls.test_results.dump()
+
+    def test_series_str_len(self):
+        pyfunc = usecase_series_len
+        hpat_func = hpat.jit(pyfunc)
+
+        for data_length, data_width in itertools.product(self.total_data_length, self.width):
+            data = perf_data_gen_fixed_len(STRIP_CASES, data_width, data_length)
+            test_data = pd.Series(data)
+
+            compile_results = calc_compilation(pyfunc, test_data)
+            # Warming up
+            hpat_func(test_data)
+
+            exec_times, boxing_times = calc_results(hpat_func, test_data)
+            self.test_results.add('series_str_len', 'JIT', test_data.size, data_width, exec_times,
+                                  boxing_times, compile_results=compile_results, num_threads=self.num_thread)
+            exec_times, _ = calc_results(pyfunc, test_data)
+            self.test_results.add('series_str_len', 'Reference', test_data.size, data_width, exec_times,
+                                  num_threads=self.num_thread)
+
+
+    def test_series_str_capitalize(self):
+        pyfunc = usecase_series_capitalize
+        hpat_func = hpat.jit(pyfunc)
+
+        for data_length, data_width in itertools.product(self.total_data_length, self.width):
+            data = perf_data_gen_fixed_len(STRIP_CASES, data_width, data_length)
+            test_data = pd.Series(data)
+
+            compile_results = calc_compilation(pyfunc, test_data)
+            # Warming up
+            hpat_func(test_data)
+
+            exec_times, boxing_times = calc_results(hpat_func, test_data)
+            self.test_results.add('series_str_capitalize', 'JIT', test_data.size, data_width, exec_times,
+                                  boxing_times, compile_results=compile_results, num_threads=self.num_thread)
+            exec_times, _ = calc_results(pyfunc, test_data)
+            self.test_results.add('series_str_capitalize', 'Reference', test_data.size, data_width, exec_times,
+                                  num_threads=self.num_thread)
+
+    def test_series_str_lower(self):
```

Comment:
the same issue here with a lot of code duplication. Need to think how to reduce duplication but not loose code clearness. It is always question of balance.